testapp/tests.py

```diff
@@ -60,86 +62,187 @@ def test_create_change_request(self, mock_get_snow_group_guid, mock_pysnow):
             'assignment_group': {'value': uuid.uuid4()},
             'state': '2'
         }
-        self.mock_pysnow_client.insert.return_value = fake_insert_retval
+
+        fake_resource = mock.MagicMock()
+        fake_resource.create.return_value = fake_insert_retval
+
+        self.mock_pysnow_client.resource.return_value = fake_resource
         mock_pysnow.Client.return_value = self.mock_pysnow_client
 
-        co = self.change_request_handler.create_change_request(
-            'title', 'description', assignment_group='assignment_group'
-        )
+        co = self.change_request_handler.create_change_request('Title', 'Description', payload={})
         last_co = ChangeRequest.objects.last()
 
-        mock_get_snow_group_guid.assert_called_with('assignment_group')
         self.assertEqual(co.pk, last_co.pk)
```

Comment:
It appears that two separate behaviors are being tested here; one to make sure that the return value of the create is as expected, and one to ensure that the object is added to the db. If that is the case, these should probably be split into separate tests.

Reply:
With the line 71 gone, it focuses only on checking the values of the persistence object created.